ui/admin-portal/src/app/components/candidates/show/show-candidates.component.ts

```diff
@@ -314,6 +325,19 @@ export class ShowCandidatesComponent implements OnInit, OnChanges, OnDestroy {
     this.localStorageService.set(this.savedListStateKey() + this.showClosedOppsSuffix, showClosedOppsValue.toString());
   }
 
+  /**
+   * Restores candidate profile search card to previous px distance from top if > 0.
+   */
+  public setSearchCardScrollTop() {
+    // Starting at 0 is default behaviour so no action needed in that case.
+    if (this.searchCardScrollTop > 0) {
+      // The card has to fully render before scrolling, so a short delay is necessary.
```

Comment:
This is a reasonable workaround. I wonder if using one of the Angular lifecycle hooks e.g. ngAfterViewInit or ngOncChanges would avoid the need for a timeout? It might be worth a look, but it is only worth spending too much time on if it doesn't give a quick result.

Reply:
@sadatmalik AfterViewInit runs _once_ after initialisation of the component — but we need these methods to run every time a new candidate is selected (which doesn't reinitialise the component, just changes its content). OnChanges runs before the view is fully rendered (as soon the candidate @Input property is changed), so can't be used to set the scrollbar position. That will be why whoever wrote the tab selection code also went for AfterViewChecked — which admittedly looked questionable to me too, at first. But I definitely saw the logic after exhausting all other possibilities!